pkg/util/kube/storage.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package kube
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	corev1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	pkgcfg "github.com/vmware-tanzu/vm-operator/pkg/config"
+	"github.com/vmware-tanzu/vm-operator/pkg/util/kube/internal"
+)
+
+// GetStoragePolicyID returns the storage policy ID for a given StorageClass.
+// If no ID is found, an error is returned.
+func GetStoragePolicyID(obj storagev1.StorageClass) (string, error) {
+	policyID, ok := obj.Parameters[internal.StoragePolicyIDParameter]
+	if !ok {
+		return "", fmt.Errorf(
+			"StorageClass %q does not have '"+
+				internal.StoragePolicyIDParameter+"' parameter",
+			obj.Name)
+	}
+	return policyID, nil
+}
+
+// MarkEncryptedStorageClass records the provided StorageClass as encrypted.
+func MarkEncryptedStorageClass(
+	ctx context.Context,
+	k8sClient ctrlclient.Client,
+	storageClass storagev1.StorageClass,
+	encrypted bool) error {
```

Comment:
Because nothing stops someone from modifying the underlying Storage Policy to no longer support encryption. This function is called by the StorageClass controller that ensures we always represent the current state of the underlying policy.